From f408b24d8d8f5b5f4e1a25c1046c3a18107c8d80 Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Wed, 4 May 2022 15:41:54 +0200
Subject: [PATCH] lib: rename VALID_FORMAT_IFACE to VALID_FORMAT

We no longer use VALID_FORMAT_IFACE for validating "iface"; rename the
macro to reflect that we only check "format" with it.

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1844341
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
Message-Id: <20220504134155.11832-4-lersek@redhat.com>
Reviewed-by: Richard W.M. Jones <rjones@redhat.com>
(cherry picked from commit c8e3caf9e6000ea2f5cfbe30ffe1240317bb4578)
---
 lib/drives.c     |  4 ++--
 lib/unit-tests.c | 16 ++++++++--------
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/lib/drives.c b/lib/drives.c
index 8fe46a41c..c5a208468 100644
--- a/lib/drives.c
+++ b/lib/drives.c
@@ -593,7 +593,7 @@ guestfs_int_free_drives (guestfs_h *g)
  * Check string parameter matches regular expression
  * C<^[-_[:alnum:]]+$> (in C locale).
  */
-#define VALID_FORMAT_IFACE(str) \
+#define VALID_FORMAT(str) \
   guestfs_int_string_is_valid ((str), 1, 0, \
                                VALID_FLAG_ALPHA|VALID_FLAG_DIGIT, "-_")
 
@@ -790,7 +790,7 @@ guestfs_impl_add_drive_opts (guestfs_h *g, const char *filename,
     return -1;
   }
 
-  if (data.format && !VALID_FORMAT_IFACE (data.format)) {
+  if (data.format && !VALID_FORMAT (data.format)) {
     error (g, _("%s parameter is empty or contains disallowed characters"),
            "format");
     free_drive_servers (data.servers, data.nr_servers);
diff --git a/lib/unit-tests.c b/lib/unit-tests.c
index 62457ccba..0e550cb98 100644
--- a/lib/unit-tests.c
+++ b/lib/unit-tests.c
@@ -434,7 +434,7 @@ test_stringsbuf (void)
 }
 
 /* Use the same macros as in lib/drives.c */
-#define VALID_FORMAT_IFACE(str) \
+#define VALID_FORMAT(str) \
   guestfs_int_string_is_valid ((str), 1, 0, \
                                VALID_FLAG_ALPHA|VALID_FLAG_DIGIT, "-_")
 #define VALID_DISK_LABEL(str) \
@@ -446,18 +446,18 @@ test_stringsbuf (void)
 static void
 test_valid (void)
 {
-  assert (!VALID_FORMAT_IFACE (""));
+  assert (!VALID_FORMAT (""));
   assert (!VALID_DISK_LABEL (""));
   assert (!VALID_HOSTNAME (""));
 
   assert (!VALID_DISK_LABEL ("012345678901234567890"));
 
-  assert (VALID_FORMAT_IFACE ("abc"));
-  assert (VALID_FORMAT_IFACE ("ABC"));
-  assert (VALID_FORMAT_IFACE ("abc123"));
-  assert (VALID_FORMAT_IFACE ("abc123-"));
-  assert (VALID_FORMAT_IFACE ("abc123_"));
-  assert (!VALID_FORMAT_IFACE ("abc123."));
+  assert (VALID_FORMAT ("abc"));
+  assert (VALID_FORMAT ("ABC"));
+  assert (VALID_FORMAT ("abc123"));
+  assert (VALID_FORMAT ("abc123-"));
+  assert (VALID_FORMAT ("abc123_"));
+  assert (!VALID_FORMAT ("abc123."));
 
   assert (VALID_DISK_LABEL ("abc"));
   assert (VALID_DISK_LABEL ("ABC"));
-- 
2.31.1

